### ISSUE TITLE
Why does loss need to be divided by num

### ISSUE THREAD

**Issue by pangda72** [CN-zdy/MSU_Net#16]
Hi CN-zdy
Thank you for your code .But i am little confused about loss.Colud you give me hints?Thank you very much!
In the "loss.py"
#==================================================
class DiceLoss(torch.nn.Module):
    def __init__(self):
        super(DiceLoss, self).__init__()

    def forward(self, SR, GT):
        num = GT.size(0)
        smooth = 1

        m1 = SR
        m2 = GT
        intersection = (m1 * m2)

        loss = 1 - (2. * intersection.sum(1) + smooth) / (m1.sum(1) + m2.sum(1) + smooth)
        loss = loss.sum() / num

        return loss
#==============================================================
I don't know why loss need to be divided by num?If I define other loss,it also need to be divided by num?
Can I comment out these two lines of code:
 num = GT.size(0)
 loss = loss.sum() / num
Hope to hear from you !
Thanks in advance.
Best.

**Comment by CN-zdy** [comment]
Hi, Pangda.
This is to find the average loss of each batch.

**Comment by pangda72** [comment]
> Hi, Pangda.
> This is to find the average loss of each batch.

Hi ,CN-zdy
Thank you for answering my questions.I got it.
